tracing/src/macros.rs

```diff
@@ -783,35 +783,52 @@ macro_rules! event {
     );
 }
 
-/// Check's whether an equivalent `event!` invocation would result in an enabled
-/// event.
+/// Checks whether a span or event is enabled depending on the provided metadata.
 ///
-/// If you are using this macro to guard a log line that requires expensive computation, it can
-/// result in false-negative's, if the default collector has filters based on line numbers or field
-/// names.
+/// This macro is a specialized tool: it is intended to be used prior
+/// to an expensive computation required *just* for that event, but
+/// *cannot* be done as part of an argument to that event, such as
+/// when multiple events are emitted (e.g., iterating over a collection
+/// and emitting an event for each item).
 ///
-/// This macro operates similarly to [`event!`], with some extensions:
-/// - Allows passing just a level.
-/// - Allows passing just a level and a target.
-///
-/// See [the top-level documentation][lib] for details on the syntax accepted by
-/// this macro.
-///
-/// [lib]: crate#using-the-macros
-/// [`event!`]: event!
+/// While `enabled!` can be used as a migration aid from [`log::log_enabled`],
+/// it is recommended that it be used sparingly. In most cases, a centralized
+/// filtering approach such as [`Targets`], [`filter_fn`], or [`EnvFilter`] will provide
+/// similar performance and greater ease-of-use. Similarly, while `enabled!` supports
```

Comment:
i don't understand this? `enabled!` _asks_ the filter (which could be an `EnvFilter`, `FilterFn`, etc) if it would enable a span/event with particular metadata. it's not a replacement for filtering?